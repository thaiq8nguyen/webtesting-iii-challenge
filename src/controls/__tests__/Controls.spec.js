import React from "react";
import { cleanup, fireEvent, render } from "@testing-library/react";

import Controls from "../Controls";

afterEach(cleanup);

/* SNAPSHOT TESTING */
test("creating controller state snapshot using snapshot testing", () => {
  const { asFragment } = render(<Controls locked={false} closed={false} />);
  expect(asFragment()).toMatchSnapshot();
});

/* UNIT TESTING */
test("lock gate button state when the close gate button is clicked", () => {
  const { getByText } = render(<Controls locked={false} closed={false} />);

  const toggleLockGateButton = getByText("Lock Gate");
  const toggleCloseGateButton = getByText("Close Gate");

  expect(toggleLockGateButton.disabled).toBeTruthy();
  expect(toggleCloseGateButton.disabled).toBeFalsy();
});
