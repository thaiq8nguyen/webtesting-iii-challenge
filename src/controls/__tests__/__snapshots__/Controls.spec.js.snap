// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creating controller state snapshot using snapshot testing 1`] = `
<DocumentFragment>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      data-testid="toggle-lock-gate"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
      data-testid="toggle-close-gate"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;
